src/Symfony/Component/Messenger/Transport/Serialization/Serializer.php

```diff
@@ -81,8 +83,10 @@ public function decode(array $encodedEnvelope): Envelope
 
         try {
             $message = $this->serializer->deserialize($encodedEnvelope['body'], $encodedEnvelope['headers']['type'], $this->format, $context);
+        } catch (NotNormalizableValueException $e) {
+            throw new MessageDecodingFailedException('No denormalizer found to decode message. Try running "composer require symfony/property-access".', $e->getCode(), $e);
```

Comment:
I was picturing that we actually just *required* property-access to be installed. Like, in the if statement above around line 53, we also check for some class in property-access and throw an exception if it's not there. That solution is simpler, your solution is fancier. If we go your direction (not sure which is better), we should at least check to see if property-access exists before throwing an exception saying to install it. I'm also not convinced (but not sure) that we need a new exception class for this - it feels strange to allow someone to "catch" this exception, when I think it will always be a misconfiguration/bug.

Reply:
I agree with not needing an extra Exception.
But: Throwing a `MessageDecodingFailedException` when encoding is worse.
We could use a more generic Exception. Which do you think would fit best?

Just checking for the existence of `PropertyAccessor` itself (like it is done for `Serializer`) seems like a bad idea, because one could technically write their own normalizer and not have `PropertyAccessor` installed, making the code work properly.
The only reason this case would happen is if no (de)normalizer installed supports objects. And that's only the case if `symfony/serializer` is installed and `symfony/property-access` isn't (as that would actively disable the `ObjectNormalizer` in the serializer compiler pass). And no other (de)normalizer supports objects. In that case, this would be an out-of-the-box solution to it.

The situation where no suported normalizers exist AND `symfony/property-access` is installed is never possible.

Reply:
I'm not sure checking specifically for it would add any value, though. The exception will never be thrown if it exists.

Reply:
Changed the exception thrown to `Symfony\Component\Messenger\Exception\LogicException` (the same one is thrown in the constructor when `Serializer` is missing).

Reply:
Ah, trying to decode a type that doesn't exist also throws the same exception. But that shouldn't trigger this extra exception message.

Reply:
I updated the code to check for both an exception message (as this is currently the only way to check the specific missing normalizer) and the presence of `PropertyAccessor`.

Is this the correct way to go? Should I also implement this in the encoding methods?

Thanks for all your feedback. 👍